### ISSUE TITLE
Investigate CSS framework replacement for website

### ISSUE THREAD

**Issue by hellais** [ooni/ooni.org#343]
We should look into replacing the CSS used by the website which is currently custom

**Comment by timepigeon** [comment]
Assuming that this is mostly about a responsive grid system and not so much about a collection of pre-designed UI elements, my vote would go to a minimal (grid-only) version of bootstrap: https://getbootstrap.com/docs/4.4/getting-started/contents/#css-files
It's widely adopted and tested, has adequate documentation and has been the most reliable I've used in the past overall (I also find the syntax preferable to more verbose options like Semantic UI or Foundation, but that's a matter of taste).

**Comment by hellais** [comment]
@timepigeon thanks for the input on this. I wasn't aware of a bootstrap version which is a grid only. We had initially excluded bootstrap because it's way too bloated and forces you to use their javascript stuff.

I will look into the bootstrap grid solution. It would also be nice to have some basic support for good typography and some very basic styling of links and buttons, but I suppose we can also do that ourselves.

Here are some notes we took during a meeting we had with @sarathms to discuss the topic with some of the options we looked at:

Pure
++ Written in CSS
++ Modular (import what you need)
++ Most popular on github
-- not flexbox, not css-grid
-- not actively maintained
https://purecss.io

Bulma
++ Flexbox based
++ Modular
++ Has react-components companion library
++ active community
-- Based on SASS
http://bulma.io/

CSS Grid
++ Modern
++ Native CSS3
-- No frameworks, needs low-level work
?? Integration with other frameworks
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout

Mini.css
++ actively developed
++ modular, can import what you need
-- Based on SASS
-- Not so mini
https://minicss.org

**Comment by timepigeon** [comment]
Hey, I've noticed the usage of the word 'sprint' lately and was asking agrabeli if you guys are doing scrum now, who refered me to you - will sprints be coordinated with community efforts in some way (e.g. reflected in github issues or sprint reviews being communicated in community meetings)? Just thinking if you guys prioritize internally it might make sense to coordinate that with community to minimise chaos (unless that's already being done / planned for)

From the ones you listed I've only used CSS Grids and Bulma (which also comes with a precompiled CSS version btw, despite being based on SASS), overall I don't think most options differ much in terms of a functioning grid system, the points for consideration I'd focus on would be:

1. probability of longterm support / development
2. compatibilty with the existing code base (e.g. UI elements included in a framework not overwriting/conflicting with the existing ooni website)
3. limiting bloat/redundancy
4. preferred style of syntax (for the people who most deal with the html markup)
5. size / performance

I can't speak much for 4.), but CSS Grids would do well on all other points, as would Bootstrap Grid-only (size wise it's at 52KB, Bulma at 192KB).
For all other options that don't offer a grid-only version 2.) could be a problem (there's a way to generate a grid-only file manually for Bulma but it's a bit hacky and probably has to be repeated after each update, I can't speak for the others).

**Comment by timepigeon** [comment]
From the ones you listed I've only used CSS Grids and Bulma (which also comes with a precompiled CSS version btw, despite being based on SASS), overall I don't think most options differ much in terms of a functioning grid system, the points for consideration I'd focus on would be:

1. probability of longterm support / development
2. compatibilty with the existing code base (e.g. UI elements included in a framework not overwriting/conflicting with the existing ooni website)
3. limiting bloat/redundancy
4. preferred style of syntax (for the people who most deal with the html markup)
5. size / performance

I can't speak much for 4.), but CSS Grids would do well on all other points, as would Bootstrap Grid-only (size wise it's at 52KB, Bulma at 192KB).
For all other options that don't offer a grid-only version 2.) could be a problem (there's a way to generate a grid-only file manually for Bulma but it's a bit hacky and probably has to be repeated after each update, I can't speak for the others).

**Comment by hellais** [comment]
Thanks for sharing these considerations. I think CSS Grids are maybe still a bit too bleeding edge (https://caniuse.com/#feat=css-grid 8% of browsers don't support them).

I am going to try out the grid only bootstrap version and see how easy it is to use that as a drop-in replacement for our existing layout system.

**Comment by hellais** [comment]
I have managed to replace the entirety of the grid layout of our website using the bootstrap grid and it was pretty seamless actually.

It would be great if folks could take a look at it and check how the site looks and feels. 

There are some minor differences with it, such as having swapped out the proprietary font we were using and some minor adjustments to the font sizing and layout.
I also added the footer from OONI Explorer to every page.

I think we should land that PR as this would also unblock other SEO related fixes to the website.
